FWCore/Framework/test/global_module_t.cppunit.cc

```diff
@@ -244,15 +245,24 @@ class testGlobalModule : public CppUnit::TestFixture {
 
     void streamEndRunSummary(edm::StreamID, edm::Run const&, edm::EventSetup const&, int*) const override { ++m_count; }
 
-    void globalEndRunSummary(edm::Run const&, edm::EventSetup const&, int*) const override { ++m_count; }
+    void globalEndRunSummary(edm::Run const&, edm::EventSetup const&, int*) const override {
+      ++m_count;
+      CPPUNIT_ASSERT(m_globalEndRunSummaryCalled == false);
+      m_globalEndRunSummaryCalled = true;
```

Comment:
@Dr15Jones Maybe we should consider some better mechanism for the call order checks, and also extend the check to cover also other cases?

Reply:
I'm all for improved testing.

Reply:
I would leave the "better mechanism" for a later PR.